src/App.jsx

```diff
@@ -1,248 +1,260 @@
+                <span className="icon is-left">
+                  <i className="fas fa-search" aria-hidden="true" />
                 </span>
-              </th>
 
-              <th>
-                <span className="is-flex is-flex-wrap-nowrap">
-                  User
-
-                  <a href="#/">
-                    <span className="icon">
-                      <i data-cy="SortIcon" className="fas fa-sort" />
-                    </span>
-                  </a>
-                </span>
-              </th>
-            </tr>
-          </thead>
-
-          <tbody>
-            <tr data-cy="Product">
-              <td className="has-text-weight-bold" data-cy="ProductId">
-                1
-              </td>
-
-              <td data-cy="ProductName">Milk</td>
-              <td data-cy="ProductCategory">🍺 - Drinks</td>
-
-              <td
-                data-cy="ProductUser"
-                className="has-text-link"
+                {(searchValue !== '') && (
+                  <span className="icon is-right">
+                    <button
+                      data-cy="ClearButton"
+                      type="button"
+                      className="delete"
+                      onClick={() => setSearchValue('')}
+                    />
+                  </span>
+                )}
+              </p>
+            </div>
+
+            <div className="panel-block is-flex-wrap-wrap">
+              <a
+                href="#/"
+                data-cy="AllCategories"
+                className={cn(
+                  'button',
+                  'is-success',
+                  'mr-6',
+                  'is-outlined',
+                  { 'is-outlined': selectedCategory.length < 0 },
+                )}
+                onClick={() => setSelectedCategory([])}
               >
-                Max
-              </td>
-            </tr>
-
-            <tr data-cy="Product">
-              <td className="has-text-weight-bold" data-cy="ProductId">
-                2
-              </td>
-
-              <td data-cy="ProductName">Bread</td>
-              <td data-cy="ProductCategory">🍞 - Grocery</td>
-
-              <td
-                data-cy="ProductUser"
-                className="has-text-danger"
+                All
+              </a>
+              {categoriesFromServer.map(category => (
+                <a
+                  data-cy="Category"
+                  href="#/"
+                  key={category.id}
+                  className={cn(
+                    'button',
+                    'mr-2',
+                    'my-1',
+                    'is-info',
+                    { 'is-info': selectedCategory.includes(category.title) },
+                  )}
+                  onClick={() => setSelectedCategory((categories) => {
+                    const index = selectedCategory.indexOf(category.title);
+
+                    return (index > -1)
+                      ? categories.slice(index, 1)
```

Comment:
Are you sure that you're using `slice` method correctly here?
https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/slice#end